test/test_other.py

```diff
@@ -890,6 +890,9 @@ def test_pkg_config_packages(self):
         out = self.run_process([emmake, 'pkg-config', '--modversion', package], stdout=PIPE).stdout
         self.assertContained(version, out)
 
+  def test_cmake_check_type_size(self):
+    self.run_process([EMCMAKE, 'cmake', test_file('cmake/check_type_size')])
```

Comment:
It's not clear immediately - does this test against something that returns the problematic error code from Node.js?

Reply:
Yes, any value larger then 255 will cause a problem.  Its not just node, native linux also can't return values larger than 255 from a process.

Reply:
I meant maybe adding a check that verifies small-but-invalid error codes too (e.g. by passing invalid flag to Node.js so that it returns `7`).

Reply:
At least for CheckSizeType that is no longer an issue right?  Since we no longer run node at all.

For other checks where we do run node itself could fail (for example with invalid flags) and we can't distinguish between the node error code and the error code of the program being run by node.   This change doesn't resolve that but also doesn't try to.    Do you have any ideas on how we might solve that issue?   Maybe we could somehow turn node error conditions into something that looks like SEGFAULT.. to avoid them looking like program return codes?

Luckily I don't think its common for autoconf or cmake tests to use non-zero program return codes to return information.. in fact I don't know of any (once we fix our hacked version CheckSizeType).  Do you?

Reply:
> At least for CheckSizeType that is no longer an issue right? Since we no longer run node at all.

Hm good point.

Overall I'm just a bit concerned about not having regression test for the issue where we were getting invalid code propagated all the way through as a sizeof. Testing large sizes is not exactly the same, because it still assumes that the program compiled successfully in the first place.

But, if there's no good way to test that after changes, I guess it's fine to go as-is.

Reply:
Yes, they seem like separable issues.   Since I've not addressed the second issue it makes sense that this change doesn't include a regression test for that.

I'm actually tempted to leave the second issue unsolved (for now), unless we can find another example where distinguishing non-zero return codes from node vs. non-zero return code from user code is important.

Reply:
Ah I thought this fixes both of those due to scanning the binary now?

Reply:
It does avoid running into the issue for CheckSizeType specifically, but if any other test was to use a non-zero return value to transmit information that method still won't work.

Oh, I think I see what you mean by adding a regression test!    You want make sure this test passes, even when node is not runnable... I can do that.

Reply:
Done

Reply:
That's perfect, thanks!